spyder/plugins/pylint/widgets/pylintgui.py

```diff
@@ -258,12 +260,36 @@ def set_filename(self, filename):
         filename = to_text_string(filename) # filename is a QString instance
         self.kill_if_running()
         index, _data = self.get_data(filename)
-        if index is None:
-            self.filecombo.addItem(filename)
-            self.filecombo.setCurrentIndex(self.filecombo.count()-1)
+        if filename not in self.curr_filenames:
+            if index is None:
+                self.filecombo.insertItem(0, filename)
+                self.filecombo.setCurrentIndex(0)
+            else:
+                self.filecombo.setCurrentIndex(
+                    self.filecombo.findText(filename))
+            self.curr_filenames.append(filename)
+
+            is_parent = self.parent is not None
+            num_elements = self.filecombo.count()
+            if is_parent and (num_elements >
+                              self.parent.get_option('max_entries')):
+                self.filecombo.removeItem(num_elements - 1)
+                self.curr_filenames.pop(num_elements - 1)
+            self.filecombo.selected()
         else:
-            self.filecombo.setCurrentIndex(self.filecombo.findText(filename))
-        self.filecombo.selected()
+            self.filecombo.setCurrentIndex(
+                self.filecombo.findText(filename))
+
+    def change_history_limit(self, new_limit):
+        """Change the number of files listed in the history combobox."""
+        if self.filecombo.count() > new_limit:
+            num_elements = self.filecombo.count()
+            diff = num_elements - new_limit
+            for i in range(diff):

```

Comment:
```suggestion
            for __ in range(diff):
```

More clearly indicate this variable is unused.